mlir/utils/performance/tuningRunner.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+
+from perfCommonUtils import Operation
+import getopt
+import subprocess
+import sys
+from pathlib import Path
+import argparse
+
+from typing import Optional
+import perfRunner
+from perfRunner import PerfConfiguration
+from perfRunner import ConvConfiguration
+from perfRunner import GemmConfiguration
+from perfRunner import Paths
+
+# global variables.
+BENCHMARKING_RESULT_FILE_NAME = 'results.stats.csv'
+
+#Tune MIOpen with MLIR kernels
+def tuneMLIRKernels(configs, paths: Paths, arch):
+    solver_names = {testVector : getSolverName(testVector, arch) for testVector in configs}
+
+    envs = os.environ.copy()
+    envs['MIOPEN_FIND_ENFORCE'] = '4'
+    envs['MIOPEN_DRIVER_USE_GPU_REFERENCE'] = '1'
+    for testVector in configs:
+        envs['MIOPEN_DEBUG_FIND_ONLY_SOLVER']=solver_names[testVector]
+        commandLine = testVector.split(sep=' ')
+        config = ConvConfiguration.fromCommandLine(commandLine, arch)
+        if config.inputLayout == 'nchw':
+            MIOpenDriverCommand = [paths.miopen_driver_path, *commandLine,'-V', '0']
+            print(' '.join(MIOpenDriverCommand))
+            p1 = subprocess.Popen(MIOpenDriverCommand, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=envs)
+            # get output.
+            try:
+               outs, errs = p1.communicate(timeout=300)
+            except subprocess.TimeoutExpired:
+                p1.kill()
+                print("MIOpen tuning timed out")
+                outs, errs = p1.communicate()
+
+def tuneGemmKernels(configs, paths: Paths, arch, debug):
+    best_perf_config = ''
+    best_ns = 0
```

Comment:
For our future sanity, please, instead of maximizing like this
- Create a `pd.DataFrame` containing the perf config and the ns. And record the inapplicable configs, either of to the side or as `np.nan`
- After the tuning loop, take the minimum of nanoseconds (ignoring `nan`)